framework/src/outputs/Checkpoint.C

```diff
@@ -62,12 +61,31 @@ Checkpoint::Checkpoint(const InputParameters & parameters)
     _num_files(getParam<unsigned int>("num_files")),
     _suffix(getParam<std::string>("suffix"))
 {
-  // Prevent the checkpoint from executing at any time other than TIMESTEP_END
+  // Prevent the checkpoint from executing at any time other than INITIAL and/or
+  // TIMESTEP_END
   const auto & execute_on = getParam<ExecFlagEnum>("execute_on");
-  if (!(execute_on.size() == 1 && execute_on.contains(EXEC_TIMESTEP_END)))
+
+  // Create a vector containing all valid values of execute_on
+  std::vector<ExecFlagEnum> valid_execute_on_values(3);
+  {
+    ExecFlagEnum valid_execute_on_value = execute_on;
+    valid_execute_on_value.clear();
+    valid_execute_on_value += EXEC_INITIAL;
+    valid_execute_on_values[0] = valid_execute_on_value;
+    valid_execute_on_value += EXEC_TIMESTEP_END;
+    valid_execute_on_values[1] = valid_execute_on_value;
+    valid_execute_on_value.clear();
+    valid_execute_on_value += EXEC_TIMESTEP_END;
+    valid_execute_on_values[2] = valid_execute_on_value;
+  }
```

Comment:
Not a huge fan of the construction of this vector. Would it make sense to implement a constructor for ExecFlagEnum (or one of its parents) so we could do something like
`ExecFlagEnum valid_execute_on_value = {EXEC_INITIAL}`?

Also, since `valid_execute_on_values` should be const after it is constructed, and will be the same for any Checkpoint object, should we find a way to make it a static const attribute?